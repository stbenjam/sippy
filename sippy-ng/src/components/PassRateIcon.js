import React from 'react'
import { Tooltip } from '@material-ui/core'
import ArrowDownwardRoundedIcon from '@material-ui/icons/ArrowDownwardRounded'
import ArrowUpwardRoundedIcon from '@material-ui/icons/ArrowUpwardRounded'
import SyncAltRoundedIcon from '@material-ui/icons/SyncAltRounded'

/**
 * PassRateIcon returns an up, down, or side-to-side arrows
 * indicating whether something improved, regressed, or stayed
 * the same.
 */
export default function PassRateIcon (props) {
  let icon = ''

  if (Math.abs(props.improvement) <= 2) {
    icon = <SyncAltRoundedIcon data-icon="SyncAltRoundedIcon" style={{ color: 'grey' }} />
  } else if (props.improvement >= 2) {
    icon = <ArrowUpwardRoundedIcon data-icon="ArrowUpwardRoundedIcon" style={{ stroke: 'green', strokeWidth: 3, color: 'green' }} />
  } else {
    icon = <ArrowDownwardRoundedIcon data-icon="ArrowDownwardRoundedIcon" style={{ stroke: 'darkred', strokeWidth: 3, color: 'darkred' }} />
  }

  if (props.tooltip) {
    return <Tooltip title={props.improvement.toFixed(2) + '%'}>{icon}</Tooltip>
  } else {
    return icon
  }
}

PassRateIcon.defaultProps = {
  tooltip: false
}
