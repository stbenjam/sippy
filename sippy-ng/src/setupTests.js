/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Polly } from '@pollyjs/core'
import HttpAdapter from '@pollyjs/adapter-node-http'
import FSPersister from '@pollyjs/persister-fs'
import fetch from 'node-fetch'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => wrapper.html().replace(/id="mui-[0-9]*"/g, '').replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')

Enzyme.configure({ adapter: new Adapter() })

global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()
