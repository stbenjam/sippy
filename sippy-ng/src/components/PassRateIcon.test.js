import React from 'react'
import { act } from '@testing-library/react'
import toJson from 'enzyme-to-json'
import { withoutMuiID } from '../setupTests'
import { mount } from 'enzyme'
import PassRateIcon from './PassRateIcon'

jest.useRealTimers()

describe(PassRateIcon, () => {
  it('shows sync icon for nearly no change', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(<PassRateIcon improvement={1} />)
    })
    expect(wrapper.find('svg[data-icon="SyncAltRoundedIcon"]').exists()).toBe(true)
    expect(toJson(withoutMuiID(wrapper))).toMatchSnapshot()
  })

  it('shows up arrow for improvement', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(<PassRateIcon improvement={3} />)
    })
    expect(wrapper.find('svg[data-icon="ArrowUpwardRoundedIcon"]').exists()).toBe(true)
    expect(toJson(withoutMuiID(wrapper))).toMatchSnapshot()
  })

  it('shows down arrow for regression', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(<PassRateIcon improvement={-3} />)
    })
    expect(wrapper.find('svg[data-icon="ArrowDownwardRoundedIcon"]').exists()).toBe(true)
    expect(toJson(withoutMuiID(wrapper))).toMatchSnapshot()
  })

  it('renders tooltip', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(<PassRateIcon improvement={-3.9999999} tooltip={true}/>)
    })

    expect(wrapper.find('ForwardRef(Tooltip)').props().title).toBe('-4.00%')
    expect(toJson(withoutMuiID(wrapper))).toMatchSnapshot()
  })
})
