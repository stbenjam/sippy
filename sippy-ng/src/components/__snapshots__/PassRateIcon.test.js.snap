// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassRateIcon renders tooltip 1`] = `undefined`;

exports[`PassRateIcon shows down arrow for regression 1`] = `undefined`;

exports[`PassRateIcon shows sync icon for nearly no change 1`] = `undefined`;

exports[`PassRateIcon shows up arrow for improvement 1`] = `undefined`;
